packages/components/src/circular-option-picker/circular-option-picker-option.tsx

```diff
@@ -46,18 +45,21 @@ function UnforwardedOptionAsOption(
 		id: string;
 		className?: string;
 		isSelected?: boolean;
-		compositeStore: NonNullable<
-			React.ComponentProps< typeof Composite >[ 'store' ]
-		>;
 	},
 	forwardedRef: ForwardedRef< any >
 ) {
-	const { id, isSelected, compositeStore, ...additionalProps } = props;
-	const activeId = useStoreState( compositeStore, 'activeId' );
+	const { id, isSelected, ...additionalProps } = props;
 
-	if ( isSelected && ! activeId ) {
-		compositeStore.setActiveId( id );
-	}
+	const { setActiveId, activeId } = useContext( CircularOptionPickerContext );
+
+	useEffect( () => {
+		if ( isSelected && ! activeId ) {
+			// The setTimeout call is necessary to make sure that this update
+			// doesn't get overridden by `Composite`'s internal logic, which picks
+			// an initial active item if one is not specifically set.
+			window.setTimeout( () => setActiveId?.( id ), 0 );
```

Comment:
https://github.com/WordPress/gutenberg/pull/64723#discussion_r1735082995

Reply:
Hey @diegohaz , do you have any advice better alternatives to using `setTimeout` here? Without it, it looks like `Composite`'s internal logic to pick an initially selected item overrides this selection.

@tyxla suggested using `queueMicrotask`, while @mirka suggesting calling `setItems` instead. (see link above). What's your take? Is this behaviour expected? Is there a more elegant way to make this work?

Reply:
After reading the comment and the discussion you linked, I still don't understand its purpose. Could you clarify?

Does Ariakit still set an active id even after you set one yourself?

Reply:
`CircularOptionPicker` displays a few color options as a composite widget.

If the component loads and then a color option is selected (programmatically), then we'd like to set that color option as the active composite item, so that when a user moves its focus on the composite widget, the selected color (and not the first color in the list) gets focus.

Reply:
An alternative that came to mind — would it be better to set the active in `onFocusVisible` ?

Reply:
> An alternative that came to mind — would it be better to set the active id in `onFocusVisible` ?

I looked into it, and it would make the code even more convoluted, since it would cause the composite wrapper to also become focusable.

What we need is:
- if the user hasn't interacted yet with the composite widget AND there is a selected option, that option should become the active item — so that focus moves to that option when the user first focuses the widget;
- in any other scenario, the first item should be the active item

@diegohaz hope the context is a bit more clear around what we're trying to achieve here

Reply:
I think this looks good to go. We can always iterate if @diegohaz has a better suggestion.

Reply:
I tried to reproduce this with Ariakit, but I might be missing something. There's no need for `setTimeout` here: https://stackblitz.com/edit/cnsby2?file=command%2Findex.tsx&theme=dark

Can you try to reproduce it from that code? This could be a bug.